agent/engine/docker_image_manager.go

```diff
@@ -56,6 +57,7 @@ type dockerImageManager struct {
 	minimumAgeBeforeDeletion         time.Duration
 	numImagesToDelete                int
 	imageCleanupTimeInterval         time.Duration
+	preservedImages                  []string
```

Comment:
Please add a a TODO here to make access to this safer without the `Unsafe` method.